include/quadrilateral_element.tcc

```diff
@@ -455,3 +455,59 @@ inline Eigen::MatrixXi
   return indices;
 }
   
+//! Return indices of a face of the element
+//! 4-noded quadrilateral
+template <>
+inline Eigen::MatrixXi
+    mpm::QuadrilateralElement<2, 4>::face_indices(unsigned face_id) const {
+  
+  //! Constants of face id and associate nodes
+  const std::map<unsigned, Eigen::Matrix<int, 2, 1>>
+      face_indices_quadrilateral{{0, Eigen::Matrix<int, 2, 1>(0, 1)},
+                                 {1, Eigen::Matrix<int, 2, 1>(1, 2)},
+                                 {2, Eigen::Matrix<int, 2, 1>(2, 3)},
+                                 {3, Eigen::Matrix<int, 2, 1>(3, 0)}}; 
+  try {
+    return face_indices_quadrilateral.at(face_id);
+  } catch (std::exception& exception) {
+    console_->error("{} #{}: {}\n", __FILE__, __LINE__, exception.what());
+  }
+}
+
+//! Return indices of a face of the element
+//! 8-noded quadrilateral
+template <>
+inline Eigen::MatrixXi
+    mpm::QuadrilateralElement<2, 8>::face_indices(unsigned face_id) const {
+  
+  //! Constants of face id and associate nodes
```

Comment:
`//! face ids and its associated nodal indices`

Reply:
Please note, in other places I have said `id` instead of ids, please use face ids in all places